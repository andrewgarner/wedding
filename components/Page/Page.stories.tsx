import { Meta, Story } from "@storybook/react";
import React from "react";

import { Page, PageProps } from "./Page";

export default {
  title: "Layout/Page",
  component: Page,
} as Meta;

const Template: Story<PageProps> = (args) => <Page {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: "The quick brown fox jumps over the lazy dog",
};
