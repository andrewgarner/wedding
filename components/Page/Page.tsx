import React, { FunctionComponent } from "react";
import { Footer } from "../Footer";
import { Header, HeaderDisplay } from "../Header";
import { Nav } from "../Nav";
import { Text, Typography } from "../Text";

interface PageProps {
  title: string;
}

const Page: FunctionComponent<PageProps> = ({ title, children }) => (
  <div className="container grid gap-4 grid-cols-1 grid-rows-page min-h-screen text-center">
    <Header display={HeaderDisplay.Invisible} />
    <Nav />
    <main className="mb-12 mx-4 opacity-0 animate-load lg:mb-16 xl:mb-24">
      <div className="mb-12 lg:mb-16 xl:mb-24">
        <Text tag="h1" typography={Typography.HeaderLarge}>
          {title}
        </Text>
      </div>
      {children}
    </main>
    <Footer />
  </div>
);

export { Page };
export type { PageProps };
