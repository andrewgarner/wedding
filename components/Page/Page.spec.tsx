import { render } from "@testing-library/react";
import React from "react";
import { Page } from "./Page";

it("renders correctly", () => {
  const { container } = render(
    <Page title="The quick brown fox jumps over the lazy dog">
      <p>Foo</p>
    </Page>
  );

  expect(container).toMatchSnapshot();
});
