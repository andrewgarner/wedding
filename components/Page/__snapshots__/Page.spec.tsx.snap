// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="container grid gap-4 grid-cols-1 grid-rows-page min-h-screen text-center"
  >
    <header
      class="invisible h-0"
    >
      <h1
        class="space-y-32 leading-loose  first-of-type:mt-0 mt-32 mb-4 mx-auto font-sans text-3xl lg:text-4xl xl:text-5xl font-normal uppercase"
      >
        <a
          href="/"
        >
          Andrew and Hannah
        </a>
      </h1>
    </header>
    <nav
      class=""
    >
      <ul
        class="inline-flex flex-row uppercase space-x-4"
      >
        <li
          class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
        >
          <a
            href="/wedding"
          >
            Wedding
          </a>
        </li>
        <li
          class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
        >
          <a
            href="/travel-and-stay"
          >
            Travel & stay
          </a>
        </li>
        <li
          class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
        >
          <a
            href="/rsvp"
          >
            RSVP
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="mb-12 mx-4 opacity-0 animate-load lg:mb-16 xl:mb-24"
    >
      <div
        class="mb-12 lg:mb-16 xl:mb-24"
      >
        <h1
          class="space-y-32 leading-loose  first-of-type:mt-8 mt-32 mb-4 mx-auto font-sans text-4xl lg:text-6xl xl:text-7xl font-normal uppercase"
        >
          The quick brown fox jumps over the lazy dog
        </h1>
      </div>
      <p>
        Foo
      </p>
    </main>
    <footer
      class="mb-12"
    >
      <p
        class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
      >
        Andrew and Hannah — 
        <time
          datetime="2021-10-09"
        >
          9/10/21
        </time>
         — Nancarrow Farm
      </p>
    </footer>
  </div>
</div>
`;
